.github/workflows/eks-performance-cluster-scaling.yml

```diff
@@ -0,0 +1,140 @@
+        description: 'Desired capacity for each node group'
+        required: true
+        type: number
+        default: 500
+
+env:
+  AWS_REGION: ${{ inputs.region || 'us-west-2' }}
+  CLUSTER_NAME: ${{ inputs.cluster_name || 'eks-performance' }}
+  DESIRED_CAPACITY: ${{ inputs.desired_capacity || 500 }}
+  TERRAFORM_AWS_ASSUME_ROLE: ${{ vars.TERRAFORM_AWS_ASSUME_ROLE }}
+  TERRAFORM_AWS_ASSUME_ROLE_DURATION: 3600  # 1 hour duration
+  CWA_GITHUB_TEST_REPO_NAME: "aws/amazon-cloudwatch-agent-test"
+  CWA_GITHUB_TEST_REPO_URL: "https://github.com/aws/amazon-cloudwatch-agent-test.git"
+  CWA_GITHUB_TEST_REPO_BRANCH: "main"
+
+jobs:
+  scale-eks-cluster:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          repository: ${{ env.CWA_GITHUB_TEST_REPO_NAME }}
+          ref: ${{ env.CWA_GITHUB_TEST_REPO_BRANCH }}
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ env.TERRAFORM_AWS_ASSUME_ROLE}}
+          aws-region: ${{ inputs.region || 'us-west-2' }}
+          role-duration-seconds: ${{ env.TERRAFORM_AWS_ASSUME_ROLE_DURATION }}
+
+      - name: Install kubectl
+        uses: azure/setup-kubectl@v3
+        with:
+          version: 'latest'
+
+      - name: Update kubeconfig for EKS cluster
+        run: |
+          aws eks update-kubeconfig --name ${{ env.CLUSTER_NAME }} --region ${{ env.AWS_REGION }}
+          kubectl get nodes -o wide
+
+      - name: Scale up node groups (Sunday)
+        if: github.event.schedule == '0 9 * * 0'
+        run: |
+          echo "Starting scale UP operation with desired capacity: 500"
+          
+          for i in {1..10}; do
+            echo "Scaling node group: eks-performance-node-${i} to 500"
+            aws eks update-nodegroup-config \
+              --cluster-name eks-performance \
+              --nodegroup-name eks-performance-node-${i} \
+              --region us-west-2 \
+              --scaling-config desiredSize=500
+          
+            echo "Waiting 1 minute before scaling next node group..."
+            sleep 60
+          done
+
+      - name: Scale down node groups (Monday)
+        if: github.event.schedule == '0 21 * * 1'
+        run: |
+          echo "Starting scale DOWN operation with desired capacity: 0"
+          
+          for i in {1..10}; do
+            echo "Scaling node group: eks-performance-node-${i} to 0"
+            aws eks update-nodegroup-config \
+              --cluster-name eks-performance \
+              --nodegroup-name eks-performance-node-${i} \
+              --region us-west-2 \
+              --scaling-config desiredSize=0
+          
+            echo "Waiting 1 minute before scaling next node group..."
+            sleep 60
+          done
+
+      - name: Scale node groups (Manual)
+        if: github.event_name == 'workflow_dispatch'
```

Comment:
we might be able to use github caches to scale up and down instead of schedule if flag

Reply:
cache with the commit_id if there is another run with the same commit_id scale it down